hibernate-testing/src/main/java/org/hibernate/testing/junit4/BaseNonConfigCoreFunctionalTestCase.java

```diff
@@ -483,9 +483,12 @@ protected boolean isCleanupTestDataRequired() {
 	protected void cleanupTestData() throws Exception {
 		Session s = openSession();
 		s.beginTransaction();
-		s.createQuery( "delete from java.lang.Object" ).executeUpdate();
-		s.getTransaction().commit();
-		s.close();
+		try {
```

Comment:
Yeah you're right that would be safer. How about trying to commit instead of rollback (in the catch block) - there is a chance that it will succeed, so that sounds to me better to try it rather than surrender and rollback straight away?! If it fails, there's nothing we can do anyway. WDYT?